Source/DafnyCore/DafnyOptions.cs

```diff
@@ -634,7 +645,7 @@ public enum IncludesModes {
           return true;
 
         case "verifyAllModules":
-          VerifyAllModules = true;
+          VerificationScope = VerificationScope.Everything;
```

Comment:
Good point. And this affects the new-style `--verify-included-files` option we should keep too. I think we have to make one of the `VerificationScopes` align exactly with `--verify-included-files` so we don't break folks then.

What about making the three options:

* `RootSources` - only files directly passed to `dafny`
* `RootAndIncludedSources` - all source files directly or indirectly passed to `dafny`. This would not include `.doo` files even if provided by `--library` since they are not in fact source files - they happen to be implemented by holding onto source but this could easily change.
* `Everything` - re-verify `.doo` files as well

Note this ultimately makes `--library` not very useful in the new default mode, since it will have no effect on the verification scope and only skip compilation, which is generally unsafe.

Just to clarify, is the use case for `Everything` to support `.doo` files that were verified with incompatible options, so you can re-verify? In that case it should also put those .doo files in scope for compilation.

Reply:
I like your suggestion ! Although I would name `RootAndIncludedSources` `AllSources`

I think the use-case for `Everything ` I had in mind is that you don't trust your library artifacts, and indeed then it should also recompile. The incompatible options one is also a good one.